lte/gateway/python/magma/pipelined/rpc_servicer.py

```diff
@@ -534,6 +539,51 @@ def GetAllTableAssignments(self, request, context):
             app_name, tables in table_assignments.items()])
 
 
+    def SetSMFSessions(self, request, context):
+
+        fut = Future()
+
+        if request.state.state in [Fsm_state.CREATED, Fsm_state.ACTIVE]:
+            self._loop.call_soon_threadsafe(
+                      self.ng_update_session_flows, request, fut)
+        else:
+            self._loop.call_soon_threadsafe(
+                      self.ng_delete_session_flows, request, fut)
+
+        return fut.result()
+
+    def ng_update_session_flows (self, request,
+                                 fut: 'Future(UPFSessionContextState)') -> UPFSessionContextState:
+
+        # Convert message containing PDR to Named Tuple Rules.
+        process_pdr_rules = OrderedDict()
+        response = self._ng_servicer_app.ng_session_message_handler(request, process_pdr_rules)
+
+        # Failure in message processing return failure
+        if response.cause_info.cause_ie != CauseIE.REQUEST_ACCEPTED:
+            fut.set_result(response)
+
+        logging.info (process_pdr_rules)
```

Comment:
here and if few other places remove extra space before bracket

Also not sure how verbose we should be here if we need debug or info. I think we probably should use debug here. @pshelar

Reply:
Comments addressed.